core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -166,6 +169,25 @@ public OpenShift(OpenShiftConfig openShiftConfig) {
 		this.waiters = new OpenShiftWaiters(this);
 	}
 
+	public void setupPullSecret(String secret) {
+		setupPullSecret("xtf-pull-secret", secret);
+	}
+
+	public void setupPullSecret(String name, String secret) {
+		Secret pullSecret = new SecretBuilder()
+				.withNewMetadata()
+					.withNewName(name)
+					.addToLabels(KEEP_LABEL, "true")
+				.endMetadata()
+				.withNewType("kubernetes.io/dockerconfigjson")
+				.withData(Collections.singletonMap(".dockerconfigjson", Base64.getEncoder().encodeToString(secret.getBytes())))
+				.build();
+		secrets().createOrReplace(pullSecret);
+		serviceAccounts().withName("default").edit().addToSecrets(new ObjectReferenceBuilder().withName(pullSecret.getMetadata().getName()).build()).done();
```

Comment:
according to https://docs.openshift.com/container-platform/4.2/openshift_images/managing-images/using-image-pull-secrets.html

we should 

* add it to the "secrets" list of the builder SA
* add it to the "imagePullSecrets" list of the default SA 

oc secrets link default <pull_secret_name> --for=pull
oc secrets link builder <pull_secret_name>

creates something like this:

```
$ oc get sa default -o yaml
apiVersion: v1
imagePullSecrets:
- name: default-dockercfg-cqfww
- name: mysecret
kind: ServiceAccount
metadata:
  creationTimestamp: "2019-10-16T11:00:03Z"
  name: default
  namespace: myproject
  resourceVersion: "1794373"
  selfLink: /api/v1/namespaces/myproject/serviceaccounts/default
  uid: 1ac74c1c-f004-11e9-a321-06b56ce0f943
secrets:
- name: default-dockercfg-cqfww
- name: default-token-tcf9j



[maschmid@archie ~]$ oc get sa builder -o yaml
apiVersion: v1
imagePullSecrets:
- name: builder-dockercfg-88ss6
kind: ServiceAccount
metadata:
  creationTimestamp: "2019-10-16T11:00:03Z"
  name: builder
  namespace: myproject
  resourceVersion: "1794457"
  selfLink: /api/v1/namespaces/myproject/serviceaccounts/builder
  uid: 1ac7896e-f004-11e9-8d2d-0603d46595f1
secrets:
- name: builder-token-dlvqs
- name: builder-dockercfg-88ss6
- name: mysecret

```

Reply:
Ok, will do. Keep or remove from `deployer` SA?

Reply:
Done